burn-core/src/optim/simple/adaptor.rs

```diff
@@ -88,11 +109,18 @@ where
             let is_require_grad = tensor.is_require_grad();
             let (key, record) = self.records.remove_entry(id).unzip();
 
+            let clipped_grad = if let Some(g_clipper) = self.gradient_clipper {
+                g_clipper.clip_gradient(grad)
+            } else {
+                grad
+            };
+
             let (tensor, state) = self.optimizer.step(
                 self.lr,
                 tensor.inner(),
-                grad,
+                clipped_grad,
                 record.map(|record| O::to_device(record.into_state(), &device)),
+                self.gradient_clipper,
```

Comment:
If the adaptor is doing the clip value, I don't think we need to pass the clipper to the optimizer.

Reply:
That's a good catch I'll adjust the adaptor and adam/sgd.